Two set of genomes are examined for this project:
1) The genomes considered Mollicutes/Mycoplasma-related endobacteria (MRE)
2) Genomes of related organisms, falling within the phylum Tenericutes

Several analyses were performed on the genomes:
1) Detect gene models and other genomic features:
 - Used Prokka for this, but could not get to work recently. A webserver-based implementation is available from KBASE
 - Also ran CRT (https://www.room220.com/crt/), tRNAscan-SE (https://github.com/UCSC-LoweLab/tRNAscan-SE), Prodigal (https://github.com/hyattpd/Prodigal), and hmmsearch against rRNA hmms from (http://www.cbs.dtu.dk/services/RNAmmer/)
 - See/run the script `./scripts/features.sh`
2) All vs all blastp searches:
 - The genes show quite low conservation (sequence similarity), suggesting that the organisms might not reasonably be considered a single genus
3) Phylogenomics with ASTRAL-MP and OrthoFinder 2
 - Dependent on the approach used, MRE appear in 1 or 2 groups. If strain DvMRE is included the result may change, because of the extreme length of this branch.
 - Single-copy ortholog approaches do not work for these genomes because there are no orthogroups found in all members of the MRE.
 - Orthogroups are generated by OrthoFinder 2 in this pipeline. Interestingly, the largest orthogroup contains toll-like receptor proteins somewhat similar to invasin from Listeria. Might be intersting to see if this protein has something to do with their endohyphal lifestyle.
 4) Annotation
 - KEGG (via KofamKOALA) - Used to classify gene function with HMMs, linked to BRITE annotations.
  * Uses webserver at https://www.genome.jp/tools/kofamkoala/
  * See/run the scripts:
   ```
   ./scripts/convert_koala_to_csv.py
   ./scripts/kegg_api_pathways.py
   ./scripts/brite_json_to_csv.py
   ./KEGG_Res/KEGG_annotation_groupings.R
   ```
 - PFAM (via PfamScan) - detect protein domains via HMMs
  * Download tool from http://ftp.ebi.ac.uk/pub/databases/Pfam/Tools/
  * See/run scripts:
  ```
  ./scripts/run_pfamscan.sh
  ./scripts/convert_pfam_to_csv.sh
  ./scripts/pfam_ip_to_go.py
  ./scripts/go_by_species.py
  ./KEGG_Res/KEGG_annotation_groupings.R
  ```
 5)
